src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -104,6 +104,12 @@ void Push(StackValue value)
         {
             FatalCheck(_stackTop < _maxStack, VerifierError.StackOverflow);
 
+            if (HasPendingPrefix(Prefix.ReadOnly))
```

Comment:
I've updated the implementation for the unbox instruction to produce a readonly value. Hope that is correct.